### PR TITLE
New execution module: textfsm

### PR DESCRIPTION
**Pull request by mirceaulinic**
### What does this PR do?

This is an execution module based on [Google TextFSM](https://github.com/google/textfsm), which extracts data from unstructured text.
Not among the best practices, but sometimes turns out that parsing raw text is the most you can do when working with archaic methodologies.

### CODE REVIEW

**Comment by mirceaulinic** [comment]
Thanks for helping out @cachedout - I was just about to push the change (and I've noticed few other small inconsistencies in the doc). :-)

**Comment by cachedout** [comment]
Go Go Jenkins!

File: salt/modules/textfsm_mod.py [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
+    HAS_TEXTFSM = False
+
+try:
+    import clitable
+    HAS_CLITABLE = True
+except ImportError:
+    HAS_CLITABLE = False
+
+# Import salt modules
+import salt.utils.files
```
Comment (cachedout) [inline salt/modules/textfsm_mod.py]: You should be able to avoid this import and use the `__utils__` dictionary instead.

Comment (cachedout) [inline salt/modules/textfsm_mod.py]: @mirceaulinic Did you see the above comment?

Comment (mirceaulinic) [inline salt/modules/textfsm_mod.py]: Hi @cachedout yes, I saw your suggestion - thanks! I'll make this change too, it's in my long queue, but given that's a new module of Oxygen has a lower prio.
Don't worry, I will let you know when ready. Cheers!

**Comment by mirceaulinic** [comment]
Hi @cachedout - this PR should be ready now. I have diverged a little bit from your suggestion, but thanks to your idea I realised the import had to be improved anyway. Please see my comments in https://github.com/saltstack/salt/pull/42336/commits/d90ed050ba0744192cfb50a1e6d3292f6e5cc437.
Thanks!

**Comment by rallytime** [comment]
re-run py3

**Review by cachedout — APPROVED** [review]

**Pull request merged**